test/unit/modunit_pico_mld.c

```diff
@@ -118,16 +119,12 @@ END_TEST
 START_TEST(tc_pico_mld_report_expired)
 {
     struct mld_timer *t = PICO_ZALLOC(sizeof(struct mld_timer));
-    struct pico_ip6 zero = {{0}};
-    struct mcast_parameters p;
+    struct pico_ip6 zero = {0};
```

Comment:
After some reading, I'm also tending towards `{0}`. If I understood correctly :
Go for the `{0}`, the standard way, Then add the `'-Wno-missing-brace` for the unit tests only.
The above should have been done in the latest commits.
I did some experiments and it appears that f.e. (using GCC 6.2):
for `struct pico_ip4 a = {0}` no warnings are generated but `struct pico_ip4 a = {{0}}` gets a warning. pico_ip4 is a uint32_t and using `{{0}}` triggers the warning "Braces around scalar initializer".
`struct pico_ip6 a = {{0}}` and `struct pico_ip6 a = {0}` produce no warnings. pico_ip6 is an array of uint8_t. So it seems the warning is dependant on the datatype.
Examination of the use of `{{0}}` in the stack reveals we always use it on datatypes/structs that contain non-scalars/other structs (not sure if this is the right way to say that, hope you understand) f.e. `struct pico_dns_ns` contains the struct pico_ip4 and using `{0}` or `{{0}}` produces no warnings.
Maybe we always got lucky when compiling the stack because of this?